src/VisualStudio/IntegrationTest/TestUtilities/InProcess/VisualStudioWorkspace_InProc.cs

```diff
@@ -141,7 +142,14 @@ public void WaitForAllAsyncOperationsOrFail(TimeSpan timeout, params string[] fe
             }
             catch (Exception e)
             {
-                Environment.FailFast("Terminating test process due to unrecoverable timeout.", e);
+                var listenerProvider = GetComponentModel().DefaultExportProvider.GetExportedValue<IAsynchronousOperationListenerProvider>();
```

Comment:
is there any chance this operation fails?

Reply:
Not in integration tests. 😄